### PR TITLE
feat: support a directory of 'current' files

### LINKED ISSUE #40: Allow current.sql to be split into several files
Hello,

After working for more than 1 month with graphile migrate, I would like to be able to split a big `current.sql` into several files smaller files. In practice, it's actually more useful than named commits as I initially requested in #28.

During the initial development phase of a project, we don't want to commit/track migrations. Our schema changes a lot, so we don't want to start building a messy migration history. We'd rather share a big `current.sql` between the development team, and commit a clean initial migration when our schema is stable. 
Also during that time, we don't release anything on a production/public environment. We'd rather enjoy the flexibility of being able to reset the database (+run the seeds) every time `current.sql` changes.

It works great so far, except for the fact that `current.sql` grew to more than 2k lines, which is not great for readability. It would be useful to be able to split `current` into smaller files. For instance `custom_1_utils.sql`, `current_2_create_user.sql`, `current_3_mutations.sql`, etc.

This feature is much needed for us so I plan to work on it. If you feel it's too specific or beyond the scope of migrate I will just make a fork. But if you're interested in a PR, here's how I see the specs:

* Automatically detect between a single or multiple `current` files, do not allow both;
* Not breaking, optional addition;
* Filename for split current is `current_[int]_[description].sql`;
* `[int]` defines the order in which the sql files are run (doesn't have to be sequential, but must be >= 1 and unique)
* `[description]` is just a text without any implication;
*  On commit, it merges all `current_` files into a single sql commit. The commit file is formated in such way that it's possible to re-create all `current_` files on `uncommit`.

I didn't think too much about the format of the commit. It definitely needs to be specific enough so it doesn't clash with a user's own comments. For instance:

```sql
--! Previous:
--! Hash:
--! Merged:
--!   1: utils
--!   2: users
 
--! Part: 1 - utils
...

--! Part: 2 - users
...
```

Let me know what you think. Thanks!

### PR DESCRIPTION
**Pull request by lukaspili**
First working prototype. Will add tests once the API is finalized.
Fixes #40 

Example of current directory:

```
current/
  1.sql
  5-blah.sql
```

1.sql
```sql
drop table if exists foo cascade;
create table foo (
  names text
);
```

5-blah.sql
```sql
drop table if exists blah cascade;
create table blah (
  foo text
);
```

After running `migrate commit`

000001.sql
```sql
--! Previous: -
--! Hash: sha1:04c41adfb2f7859a0fb029efa2679a281e5b6192

--! split: 1.sql

drop table if exists foo cascade;
create table foo (
  names text
);

--! split: 5-blah.sql

drop table if exists blah cascade;
create table blah (
  foo text
);
```

`!-- split` is always named, even if there are no message, because it's needed to restore the filenames on `uncommit`. Otherwise when commiting `1.sql` and `10.sql`, it would be uncommited as `1.sql` and `2.sql`.

Right now, the logic is rather simple. Treat `current/` as a single merged `current.sql` for any operations, such as watching and re-running migrations on changes.

### CODE REVIEW

**Comment by lukaspili** [comment]
Regarding `--! no-transaction` and the readme's notice ("must contain exactly one statement"), it would make sense to not allow having multiple migration files and `--! no-transaction` at the same time.

**Comment by lukaspili** [comment]
The PR is feature complete, minus tests.

**Comment by lukaspili** [comment]
Thanks for the review @benjie. 
Need a few more days before being available to finish this.

File: src/commands/commit.ts [hunk 1]
```diff
@@ -1,17 +1,16 @@
...
 import * as fsp from "../fsp";
 import { calculateHash } from "../hash";
-import { _reset } from "./reset";
-import { _migrate } from "./migrate";
 import { logDbError } from "../instrumentation";
+import { getAllMigrations } from "../migration";
+import { ParsedSettings, parseSettings, Settings } from "../settings";
+import { _migrate } from "./migrate";
+import { _reset } from "./reset";
+import pgMinify = require("pg-minify");
```
[the remark this answers is not in the document]
Comment (lukaspili) [reply src/commands/commit.ts]: Sorry for that, it's a mistake. Will revert.

**Comment by benjie** [comment]
Sure thing 👍

**Comment by benjie** [comment]
(Just a ping to keep this on your radar, @lukaspili )

**Comment by benjie** [comment]
@lukaspili I'm going to take a look at this now

**Comment by lukaspili** [comment]
I'm sorry @benjie for not finishing the PR. 
Thanks a lot for taking it to completion!

**Comment by benjie** [comment]
No worries; it became much more complicated by the multifile changes!

**Pull request merged**